tensorbay/utility/attr.py

```diff
@@ -0,0 +1,213 @@
+            contents: A dict containing all data of internal attributes.
+
+        """
+        contents: Dict[str, Any] = {}
+        for support in getattr(self, "_supports", []):
+            contents.update(support._dumps(self))  # pylint: disable=protected-access
+
+        for name, field in self._attrs_fields.items():
+            if field.is_dynamic and not hasattr(self, name):
+                continue
+
+            if field.default is not ...:
+                value = getattr(self, name, field.default)
+            else:
+                value = getattr(self, name)
+
+            contents[field.key] = field.dumper(value)
+        return contents
+
+
+def attr(
+    *,
+    is_dynamic: bool = False,
+    key: str = "",
+    default: Any = ...,
+    error_message: Optional[str] = None,
+) -> Any:
+    """Return an instance to identify attribute fields.
+
+    Arguments:
+        is_dynamic: Determine if this is a dynamic attribute.
+        key: Display value of the attribute.
+        default: Default value of the attribute.
+        error_message: The custom error message of the attribute.
+
+    Raises:
+        AttrConflictError: When specifying both dynamic and default value.
+
+    Returns:
+        A Field instance containing all arguments.
+
+    """
+    if is_dynamic and default is not ...:
+        raise AttrConflictError()
+    return Field(is_dynamic, key, default, error_message)
+
+
+def _get_field(obj: Any, name: str, annotation: Any) -> Optional[Field]:
+
+    field = getattr(obj, name, None)
+    if not isinstance(field, Field):
+        return None
+
+    field.loader, field.dumper = _get_operators(annotation)
+    if not field.key:
+        field.key = name
+    delattr(obj, name)
+
+    return field
+
+
+def _get_operators(
+    annotation: Type[Any], is_internal: bool = False
+) -> Tuple[Callable[[Any], Any], Callable[[Any], Any]]:
+    """Get attribute operate method by annotations.
+
+     AttrsMixin has three operate types which is classified by attribute annotation.
+        1. builtins operate, like str, int, None
+        2. tensorbay custom class, like tensorbay.label.Classification
+        3. tensorbay custom class list, like List[tensorbay.label.LabeledBox2D]
+
+    Arguments:
+        annotation: Type of the attribute.
+        is_internal: Whether the annotation is inside of typing.List.
+
+    Returns:
+        Operate methods of the annotation.
+
+    """
+    if getattr(annotation, "__module__", None) in _BUILTINS:
+        return (_builtin_operator, _builtin_operator)
+
+    if getattr(annotation, "__origin__", None) == list:
+        return _get_operators(annotation.__args__[0], True)
+
+    if not is_internal:
+        return (annotation.loads, _attr_dumper)
+
+    return (
+        lambda contents: [annotation.loads(content) for content in contents],
+        _attr_list_dumper,
+    )
+
+
+def _builtin_operator(contents: _T) -> _T:
+    return contents
+
+
+def _attr_dumper(attr_: Any) -> Any:
```

Comment:
try `Protocol` instead of `Any`?
https://docs.python.org/zh-cn/3/library/typing.html#typing.Protocol